sled-hardware/src/disk.rs

```diff
@@ -64,6 +67,20 @@ pub enum DiskError {
     MissingStorageKeyRequester,
     #[error("Encrypted filesystem '{0}' missing 'oxide:epoch' property")]
     CannotParseEpochProperty(String),
+    #[error("Encrypted dataset '{dataset}' cannot set 'oxide:agent' property: {err}")]
+    CannotSetAgentProperty {
+        dataset: String,
+        #[source]
+        err: Box<zfs::SetValueError>,
+    },
+    #[error(
+        "Encrypted dataset '{dataset}' missing 'oxide:agent' property: {err}"
+    )]
+    CannotParseAgentProperty {
+        dataset: String,
+        #[source]
+        err: zfs::GetValueError,
+    },
```

Comment:
Looks like this variant is unused? (Which is probably a good thing, because otherwise if we wanted to enable wiping a dataset in a later version, it would fail to parse this property because we wouldn't have set it before.)

Reply:
Good catch, I'll remove it. I was trying to handle more of these error cases explicitly, but just decided to lump this into the "auto-destroy" case instead.